js/modules/k6/html/element.go

```diff
@@ -0,0 +1,357 @@
+	if prev := e.qsel.Prev(); prev.Length() > 0 {
+		return selToElement(Selection{e.rt, prev})
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) NextElementSibling() goja.Value {
+	if next := e.qsel.Next(); next.Length() > 0 {
+		return selToElement(Selection{e.rt, next})
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) ParentNode() goja.Value {
+	if e.node.Parent != nil {
+		return nodeToElement(e, e.node.Parent)
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) ParentElement() goja.Value {
+	if prt := e.qsel.Parent(); prt.Length() > 0 {
+		return selToElement(Selection{e.rt, prt})
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) ChildNodes() []goja.Value {
+	return elemList(e.sel.Contents())
+}
+
+func (e Element) Children() []goja.Value {
+	return elemList(e.sel.Children())
+}
+
+func (e Element) ChildElementCount() int {
+	return e.sel.Children().Size()
+}
+
+func (e Element) ClassList() []string {
+	if clsName, exists := e.qsel.Attr("class"); exists {
+		return strings.Fields(clsName)
+	} else {
+		return nil
+	}
+}
+
+func (e Element) ClassName() goja.Value {
+	return e.sel.Attr("class")
+}
+
+func (e Element) Lang() goja.Value {
+	if attr := getHtmlAttr(e.node, "lang"); attr != nil && attr.Namespace == "" {
+		return e.rt.ToValue(attr.Val)
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) OwnerDocument() goja.Value {
+	if node := getOwnerDocNode(e.node); node != nil {
+		return nodeToElement(e, node)
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (e Element) NamespaceURI() string {
+	return namespaceURI(e.node.Namespace)
+}
+
+func (e Element) IsDefaultNamespace() bool {
+	return e.node.Namespace == ""
+}
+
+func getOwnerDocNode(node *gohtml.Node) *gohtml.Node {
+	for ; node != nil; node = node.Parent {
+		if node.Type == gohtml.DocumentNode {
+			return node
+		}
+	}
+	return nil
+}
+
+func (e Element) InnerHTML() goja.Value {
+	return e.sel.Html()
+}
+
+func (e Element) NodeType() goja.Value {
+	switch e.node.Type {
+	case gohtml.TextNode:
+		return e.rt.ToValue(3)
+
+	case gohtml.DocumentNode:
+		return e.rt.ToValue(9)
+
+	case gohtml.ElementNode:
+		return e.rt.ToValue(1)
+
+	case gohtml.CommentNode:
+		return e.rt.ToValue(8)
+
+	case gohtml.DoctypeNode:
+		return e.rt.ToValue(10)
+
+	default:
+		return goja.Undefined()
+	}
+}
+
+func (e Element) NodeValue() goja.Value {
+	switch e.node.Type {
+	case gohtml.TextNode:
+		return e.rt.ToValue(e.sel.Text())
+
+	case gohtml.CommentNode:
+		return e.rt.ToValue(e.sel.Text())
+
+	default:
+		return goja.Undefined()
+	}
+}
+
+func (e Element) Contains(v goja.Value) bool {
+	if other, ok := v.Export().(Element); ok {
+		// when testing if a node contains itself, jquery + goquery Contains() return true, JS return false
+		return other.node == e.node || e.qsel.Contains(other.node)
+	} else {
+		return false
+	}
+}
+
+func (e Element) Matches(selector string) bool {
+	return e.qsel.Is(selector)
+}
+
+//helper methods
```

Comment:
These would probably look better at the top of the file or in a `util.go` file, and much easier to find.